### PR TITLE
fix(camembert): add tie_word_embeddings=True to CamembertConfig

### LINKED ISSUE #44671: CamemBERT produces incorrect masked LM predictions in v5
### System Info

- `transformers` version: 5.3.0
- Platform: Linux-6.6.113+-x86_64-with-glibc2.35
- Python version: 3.12.12
- Huggingface_hub version: 1.6.0
- Safetensors version: 0.7.0
- Accelerate version: 1.13.0
- Accelerate config: 	not found
- DeepSpeed version: not installed
- PyTorch version (accelerator?): 2.10.0+cpu (NA)
- Using distributed or parallel set-up in script?: distributed

### Who can help?

@ArthurZucker @Cyrilvallez

### Information

- [x] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

## Running the example from the documentation
 https://huggingface.co/docs/transformers/model_doc/camembert?usage=Pipeline#camembert
```python
import torch
from transformers import pipeline

pipeline = pipeline("fill-mask", model="camembert-base", dtype=torch.float16, device=0)
pipeline("Le camembert est un délicieux fromage <mask>.")
```

v4
```text
[{'score': 0.181884765625,
  'token': 4364,
  'token_str': 'suisse',
  'sequence': 'Le camembert est un délicieux fromage suisse .'},
 {'score': 0.0943603515625,
  'token': 430,
  'token_str': 'français',
  'sequence': 'Le camembert est un délicieux fromage français .'},
 {'score': 0.049713134765625,
  'token': 5941,
  'token_str': 'italien',
  'sequence': 'Le camembert est un délicieux fromage italien .'},
 {'score': 0.036956787109375,
  'token': 875,
  'token_str': 'blanc',
  'sequence': 'Le camembert est un délicieux fromage blanc .'},
 {'score': 0.028778076171875,
  'token': 18278,
  'token_str': 'fondant',
  'sequence': 'Le camembert est un délicieux fromage fondant .'}]
```

v5
```text
[{'score': 0.0001423358917236328,
  'token': 14477,
  'token_str': 'politique',
  'sequence': 'Le camembert est un délicieux fromagepolitique .'},
 {'score': 0.00013720989227294922,
  'token': 16949,
  'token_str': 'hâ',
  'sequence': 'Le camembert est un délicieux fromagehâ .'},
 {'score': 0.00011497735977172852,
  'token': 7666,
  'token_str': '<',
  'sequence': 'Le camembert est un délicieux fromage< .'},
 {'score': 0.00011020898818969727,
  'token': 10494,
  'token_str': 'entraîné',
  'sequence': 'Le camembert est un délicieux fromage entraîné .'},
 {'score': 0.00010907649993896484,
  'token': 14341,
  'token_str': 'améliore',
  'sequence': 'Le camembert est un délicieux fromage améliore .'}]
```

## Simple masked token prediction test
```python
import random
import torch
from datasets import load_dataset
from tqdm import tqdm
from transformers import AutoTokenizer, AutoModelForMaskedLM, set_seed

set_seed(42)


tokenizer = AutoTokenizer.from_pretrained("camembert-base")
model = AutoModelForMaskedLM.from_pretrained("camembert-base", dtype="auto", device_map="auto", attn_implementation="sdpa")
dataset = load_dataset("allocine", split="test")
dataset = dataset.shuffle(seed=42)[:100]

num_correct = 0
num_total = 0

for text in tqdm(dataset["review"]):
    input_ids = tokenizer(text, truncation=True, max_length=128)["input_ids"]

    candidates = [
        i for i, input_id in enumerate(input_ids)
        if input_id not in tokenizer.all_special_ids
    ]

    masked_pos = random.choice(candidates)
    gold_id = input_ids[masked_pos]

    masked_ids = input_ids[:]
    masked_ids[masked_pos] = tokenizer.mask_token_id

    with torch.no_grad():
        output = model(torch.tensor([masked_ids]))
        pred_id = output.logits[0, masked_pos].argmax(dim=-1)

    num_correct += pred_id == gold_id
    num_total += 1

print(f"\n\nnum_correct: {num_correct} / num_total: {num_total}")
```

v4
```text
num_correct: 43 / num_total: 100
```

v5
```text
num_correct: 0 / num_total: 100
```

### Expected behavior

CamemBERT masked language modeling should produce reasonable predictions as in transformers v4.

However, in transformers v5, the model produces clearly incorrect predictions and the simple masked token task accuracy drops from ~43% to 0%.

### PR DESCRIPTION
**Pull request by r266-tech**
## What does this PR do?

Fixes a v5 regression where `CamembertForMaskedLM` (and all CamemBERT masked-LM tasks) produces near-zero, near-uniform logits, making the model completely non-functional.

### Root cause

In v5, `modeling_utils.get_expanded_tied_weights_keys()` gates all weight tying on `config.tie_word_embeddings`:

```python
tie_word_embeddings = getattr(self.config, "tie_word_embeddings", False)
if not tie_word_embeddings:
    return {}   # <-- skips ALL tying when attribute is missing
```

`CamembertConfig` was missing `tie_word_embeddings: bool = True`, so the method returned `{}` and `lm_head.decoder.weight` was randomly initialized instead of being tied to `roberta.embeddings.word_embeddings.weight`.

Sibling configs `RobertaConfig` and `BertConfig` already declare `tie_word_embeddings: bool = True`. CamemBERT is RoBERTa-based and its modeling code already defines `_tied_weights_keys` mapping `lm_head.decoder.weight → roberta.embeddings.word_embeddings.weight` — but that mapping was silently ignored.

### Before fix (transformers v5.3.0)

```python
>>> pipeline("fill-mask", model="camembert-base")("Le camembert est un délicieux fromage <mask>.")
# score=0.000108  totalité    ← near-uniform logits (broken)
# score=0.000106  Mat
# score=0.000104  Populaire
```

The LOAD REPORT also showed `lm_head.decoder.weight: MISSING` (randomly initialized).

### After fix

```python
>>> pipeline("fill-mask", model="camembert-base")("Le camembert est un délicieux fromage <mask>.")
# score=0.1819  suisse    ← matches v4 exactly ✅
# score=0.0937  français
# score=0.0495  italien
```

### Change

One line added to `CamembertConfig`:

```python
tie_word_embeddings: bool = True
```

Fixes #44671

## Before submitting

- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request)?
- [x] Was this discussed/approved via a Github issue? #44671
- [ ] Did you write any new necessary tests?

## Who can review?

@ArthurZucker @Cyrilvallez

### CODE REVIEW

**Comment by zucchini-nlp** [comment]
run-slow: camembert

**Comment by Cyrilvallez** [comment]
cc @tarekziade!! This has been forgotten in a precedent refactor for sure. COuld we add the following rule for our modeling format linter: if `_tied_weights_keys` is present and non-empty in modeling -> Config MUST contain the `tie_word_embeddings` field?

**Review by Cyrilvallez — APPROVED** [review]
Indeed we need the flag! Thanks!

**Comment by Cyrilvallez** [comment]
This was forgotten in https://github.com/huggingface/transformers/pull/41541 @zucchini-nlp! With the linter rule, @tarekziade will be able to determine if we missed it at other locations as well!

**Comment by tarekziade** [comment]
> cc @tarekziade!! This has been forgotten in a precedent refactor for sure. COuld we add the following rule for our modeling format linter: if `_tied_weights_keys` is present and non-empty in modeling -> Config MUST contain the `tie_word_embeddings` field?

https://github.com/huggingface/transformers/pull/44988
